src/config-import.js

```diff
@@ -183,6 +199,60 @@ export async function readConfig(configPath) {
               ? preset.icon
               : undefined,
           value: presetValue,
+          name,
+        }
+      }
+    },
+    /**
+     * @returns {Iterable<{name: string,value:import('@mapeo/schema').TranslationValue}>}
+     */
+    *translations() {
+      if (!translationsFile) return
+      for (let [lang, languageTranslations] of Object.entries(
+        translationsFile
+      )) {
+        const { language: languageCode, region: regionCode } = parseBCP47(lang)
+        if (!languageCode) throw new Error(`invalid translation language`)
+        for (let [
+          schemaNameRef,
+          languageTranslationsForDocType,
+        ] of Object.entries(languageTranslations)) {
+          if (!isRecord(languageTranslationsForDocType))
+            throw new Error(`invalid translation docType`)
+          for (let [docName, fieldsToTranslate] of Object.entries(
+            languageTranslationsForDocType
+          )) {
+            if (!isRecord(fieldsToTranslate))
+              throw new Error(`invalid translation field`)
+            for (let [fieldRef, message] of Object.entries(fieldsToTranslate)) {
+              // we skip messages that aren't strings, for now
+              if (isRecord(message)) continue
+              if (typeof message !== 'string')
+                throw new Error(`invalid translation message`)
+              let translationValue = {
+                /** @type {'translation'} */
+                schemaName: 'translation',
+                languageCode,
+                regionCode: regionCode || '',
+                schemaNameRef,
+                fieldRef,
+                message,
+                docIdRef: '',
```

Comment:
I was surprised that `docIdRef` was an empty string here. It seems like it's necessary for types/validation but it's impossible to actually know at this point.

I wonder if we should return a slightly different type here, where `docIdRef` is omitted?

Reply:
Yeah, I'll spread the `translationValue` with `docIdRef: ''` only when doing validation, but return a type that `Omit`s that field (since it will be filled later when actually importing the config)